components/callbacks.py

```diff
@@ -446,3 +449,38 @@ async def buy(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
         await try_to_delete(message.reply_to_message)
 
     await try_to_delete(message)
+
+
+async def _token_warning(
+    update: Update, context: ContextTypes.DEFAULT_TYPE, middle_text: str = ""
+) -> None:
+    """Warn people when they share their bot's token, and tell them to revoke it"""
+    message = cast(Message, update.effective_message)
+
+    # Update timestamp on chat_data, and get "x time since last time" text
+    last_time = update_shared_token_timestamp(update, context)
+
+    # Send the message
+    await message.reply_text(f"{TOKEN_TEXT}{middle_text}Previous token was shared: {last_time}")
+
+
+async def regex_token_warning(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
+    matches = cast(List[str], context.matches)
+
+    bots = []
+    for match in matches:
+        bot = await get_bot_from_token(match)
+        if bot is not None:
+            bots.append(bot.mention_html())
+
+            # Limit to 10 bots
+            if len(bots) == 10:
+                await _token_warning(update, context, f"<b>{', '.join(bots)}</b>\n\n")
+                return
+
+    if bots:
+        await _token_warning(update, context, f"<b>{', '.join(bots)}</b>\n\n")
+
+
+async def command_token_warning(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
+    await _token_warning(update, context)
```

Comment:
I completely forgot about removing the message, yeah. 

The idea for this command is to warn users that posted a token via pastebin, the ones on a message should already get handled by the `regex` filter.
As such, getting info about the bot whose token was shared would need some web scrapping (or even image recognition if the token was shared via a code image on imgur or the like), and I don't feel like doing that much just to add the bot's name when we can already warn them.
Could also make the command receive an optional argument `/token <token>`, tho

Reply:
Ah, you're right ofc. Well, then scrape what I said :D What would be good though is a short docstring for this function (and maybe als `regex_token_warning`), i.e. just 2-3 sentences explaining what this is used for. This might avoid similar confusion in the future :)

If you like to, you can also add this new functionality to the readme :)